gnocchi/carbonara.py

```diff
@@ -85,6 +84,46 @@ def round_timestamp(ts, freq):
         (ts - UNIX_UNIVERSAL_START64) / freq) * freq
 
 
+def make_timeseries_array(array):
+    """Create a Numpy array with the right format for a timeseries."""
+    return numpy.array(array,
+                       dtype=[('timestamps', 'datetime64[ns]'),
+                              ('values', 'float64')])
+
+
+def make_timeseries(timestamps, values):
+    """Return a Numpy array representing a timeseries.
+
+    This array specifies correctly the data types, which is important for
+    Numpy to operate fastly.
+    """
+    l = len(timestamps)
+    if l != len(values):
+        raise ValueError("Timestamps and values must have the same length")
+    arr = make_timeseries_array(numpy.zeros(l))
+    arr['timestamps'] = timestamps
+    arr['values'] = values
+    return arr
+
+
+def combine_timeseries(ts1, ts2):
```

Comment:
Maybe, but overall it seems the whole code is still faster so I'm not really bothered?

I'll let @chungg yell at me if it's way slower than Pandas. :)